integration-cli/docker_cli_commit_test.go

```diff
@@ -155,3 +155,17 @@ func (s *DockerSuite) TestCommitChangeLabels(c *check.C) {
 	// check that container labels didn't change
 	c.Assert(inspectField(c, "test", "Config.Labels"), checker.Equals, "map[some:label]")
 }
+
+func (s *DockerSuite) TestCommitInheritEnv(c *check.C) {
+	dockerCmd(c, "run", "--name", "test", "busybox", "true")
+	dockerCmd(c, "commit",
+		"--change", "ENV PATH=/bin",
+		"test", "test-img-1")
+	dockerCmd(c, "run", "--name", "test2", "test-img-1", "true")
+	imageID, _ := dockerCmd(c, "commit",
+		"--change", "ENV PATH=/usr/bin:$PATH",
+		"test2", "test-img-2")
+	imageID = strings.TrimSpace(imageID)
```

Comment:
I decided to commit twice to make this test less dependent on the details of busybox (though I doubt that busybox's `PATH` is going to change any time soon)